dadi/frontend/components/DocumentFilters/DocumentFilters.jsx

```diff
@@ -19,52 +19,75 @@ export default class DocumentFilters extends Component {
     } = this.props
 
     // Evaluate passed filter / store in state
+    let filters = (isValidJSON(filter) ? objectToArray(JSON.parse(filter), 'field') : []).map(this.deconstructFilters)
     this.state = {
```

Comment:
I have updated this, but suggest we do it everywhere it we really want to treat state as immutable